### ISSUE TITLE
DSSP should return protein-only SS predictions

### ISSUE THREAD

**Issue by cing** [mdtraj/mdtraj#803]
When you run ```compute_dssp(traj)``` on a topology that has waters/junk floating around, you get an array with shape=(n_frames, n_residues) where waters/junk count as residues. I'm guessing that somewhere in the algorithm DSSP doesn't compute anything for residues without backbone atoms N, CA, C, O and so it just returns Coil for those ones. Is this true? All my waters are unanimously 'C' so... it seems like it!

Anyway, is there any reason why compute_dssp doesn't automatically run/return secondary structure predictions for protein-only residues? Seems like it could be a source of confusion.

In lieu of preprocessing your trajectory to remove waters (via laziness), I discovered the work around is to post-process your DSSP array to get the protein-only residues like so:

```
ss=md.compute_dssp(t)
rsel = [res.index for res in t.topology.residues if res.is_protein]
ss[:,rsel]
```

**Comment by rmcgibbo** [comment]
> Anyway, is there any reason why compute_dssp doesn't automatically run/return secondary structure predictions for protein-only residues?

Then we need to deal with the switch in indexing though. Like, if the protein residues are not the first `N` residues in the topology, then `dssp_result[:, i]` and `topology.residue(i)` will not correspond to the same residue.

>  Seems like it could be a source of confusion.

I agree.

Maybe we should use a different letter instead of `C` for "residues" that don't satisfy the appropriate prerequisites. Perhaps labeling these with the space character would be appropriate.

**Comment by cing** [comment]
Big fan of this solution, thanks!

**Comment by rmcgibbo** [comment]
Fixed by #841.
